### PR TITLE
lib/vfscore: Fix implicit declaration of `__lxstat`

### LINKED ISSUE #659: __uk_syscall_r_lstat build warning in lib/vfscore/main.c
### Describe the bug

When building `helloworld-cpp` with musl there is a warning around `__uk_syscall_r_lstat `

### Steps to reproduce

Using the script from [1] and running the commands:

```
./do-helloworld-cpp setup
./do-helloworld-cpp build
```

[1] https://github.com/unikraft-upb/scripts/blob/main/musl/do-helloworld-cpp

### Expected behavior

There should be no build warnings.

### Which architectures were you using or does this bug affect?

x86_64

### Which platforms were you using or does this bug affect?

kvm

### Relevant log output

```shell
/home/radu/Downloads/scripts/workdir/unikraft/lib/vfscore/main.c: In function ‘__uk_syscall_r_lstat’:
/home/radu/Downloads/scripts/workdir/unikraft/lib/vfscore/main.c:1605:9: warning: implicit declaration of function ‘__lxstat’; did you mean ‘lstat’? [-Wimplicit-function-declaration]
 1605 |  return __lxstat(1, pathname, st);
      |         ^~~~~~~~
      |         lstat
```
```

### PR DESCRIPTION
**Pull request by eduardvintila**
### Prerequisite checklist

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.pl`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.pl) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.

### Base target

 - Architecture(s): [e.g. `x86_64` or N/A]
 - Platform(s): [e.g. `kvm`, `xen` or N/A]
 - Application(s): [e.g. `app-python3` or N/A]

### Description of changes

If the `UK_LIBC_SYSCALLS` macro is not set, then `__lxstat` is to be provided by the libc. Since `__lxstat` isn't actually part of any header file, we must include a forward declaration of the function in order to avoid any compile-time warnings.

Closes #659 

Signed-off-by: Eduard Vintilă <eduard.vintila47@gmail.com>

### CODE REVIEW

**Comment by nderjung** [comment]
Hi @eduardvintila,

Thanks for this.  Can you add the following Git trailer to your commit message (before the sign-off):

```
GitHub-Fixes: #659
```

This will automatically close the issue once merged.

**Comment by razvand** [comment]
All good for me, waiting for @RaduNichita 's review. And, @eduardvintila , please see @nderjung 's comment.

**Comment by eduardvintila** [comment]
> Hi @eduardvintila,
> 
> Thanks for this. Can you add the following Git trailer to your commit message (before the sign-off):
> 
> ```
> GitHub-Fixes: #659
> ```
> 
> This will automatically close the issue once merged.

Thanks, @nderjung! I've modified my PR now.

**Review by RaduNichita — APPROVED** [review]
All good also from my side. It works and I don't see the warning anymore. Thanks, @eduardvintila for providing the fix. 

Reviewed-by: Radu Nichita [radunichita99@gmail.com](mailto:radunichita99@gmail.com)

**Review by razvand — APPROVED** [review]
Reviewed-by: Razvan Deaconescu <razvand@unikaft.io>
Approved-by: Razvan Deaconescu <razvand@unikaft.io>

**Pull request closed without merging**
